File: frontend/src/app/user/service/form.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import {Member} from '../../model/member';
import {UniversityCourses} from '../../model/university-courses';
import {UniversityDepartments} from '../../model/university-departments';
import {Project} from '../../model/project';
import {WebStarterDesc} from '../../model/message/web-starter-desc';

@Injectable({
  providedIn: 'root'
})
export class FormService {

  constructor(private http: HttpClient) { }

  public sendForm(member: Member, idProject: number): Observable<any> {
    const url = 'http://localhost:8080/member/form/' + idProject;
    return this.http.post<any>(url, member, { observe: 'response'});
   }

  public getUniversityCourses(): Observable<UniversityCourses[]> {
    const url = 'http://localhost:8080/member/university-courses';
    return this.http.get<UniversityCourses[]>(url);
  }
  public getUniversityDepartments(): Observable<UniversityDepartments[]> {
    const url = 'http://localhost:8080/member/university-departments';
    return this.http.get<UniversityDepartments[]>(url);
  }
  public getActiveProject(): Observable<Project[]> {
    const url = 'http://localhost:8080/member/active-project';
    return this.http.get<Project[]>(url);
  }
  public getWebStarterDesc(): Observable<WebStarterDesc> {
    const url = 'http://localhost:8080/member/get-web-starter';
    return this.http.get<WebStarterDesc>(url);
  }
}

File: backend/src/main/java/pl/pancerro/backend/service/formService/FormService.java
package pl.pancerro.backend.service.formService;

import pl.pancerro.backend.model.Member;
import pl.pancerro.backend.model.Project;
import pl.pancerro.backend.model.UniversityCourses;
import pl.pancerro.backend.model.UniversityDepartments;
import pl.pancerro.backend.model.message.WebStarterDesc;

import javax.mail.MessagingException;
import java.util.List;

public interface FormService {
    boolean sendForm(Member member,long idProject) throws MessagingException;
    List<UniversityCourses> getUniversityCourses();
    List<UniversityDepartments> getUniversityDepartments();
    List<Project> getActiveProject();
    WebStarterDesc getWebStarterDesc();

}
